modules/modal/react/lib/ModalContent.tsx

```diff
@@ -85,17 +81,17 @@ const Container = styled('div')({
   background: 'rgba(0,0,0,0.65)',
   animationName: `${fadeIn}`,
   animationDuration: '0.3s',
-  zIndex: 1,
 });
 
 const transformOrigin = {
   horizontal: 'center',
   vertical: 'bottom',
 } as const;
 
-function getFirstElementToFocus(modalEl: HTMLElement): HTMLElement {
-  const firstFocusable = modalEl.querySelector<HTMLElement>(
-    `[data-close=close],[id="${modalEl.getAttribute('aria-labelledby')}"]`
+function getFirstElementToFocus(overlayEl: HTMLElement): HTMLElement {
+  const modalEl = overlayEl.querySelector('[role=dialog]');
+  const firstFocusable = modalEl?.querySelector<HTMLElement>(
```

Comment:
This change was necessary because the `modalRef` is now pointing to the overlay instead of the `[role=dialog]` element